src/event_loop.rs

```diff
@@ -212,16 +212,23 @@ impl<T: 'static> fmt::Debug for EventLoopProxy<T> {
 
 /// The error that is returned when an `EventLoopProxy` attempts to wake up an `EventLoop` that
 /// no longer exists.
-#[derive(Debug, Copy, Clone, PartialEq, Eq, Hash)]
-pub struct EventLoopClosed;
+/// Optionally contains the original event given to `send_event` if the platform supports it.
+#[derive(Copy, Clone, PartialEq, Eq, Hash)]
+pub struct EventLoopClosed<T>(pub Option<T>);
 
-impl fmt::Display for EventLoopClosed {
+impl<T> fmt::Debug for EventLoopClosed<T> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        write!(f, "EventLoopClosed {{ .. }}")
+    }
+}
```

Comment:
It would require that `T` derives `Debug`, thus the `std::error::Error` impl would require `T: Debug`, too.

Reply:
I implemented it similiar to `EventLoopProxy`.

Reply:
I'd prefer you do a normal `#[derive(Debug)]` here. `EventLoopProxy` has a manual implementation because it doesn't actually contain any instances of `T`, so it never needs to print `T` out, whereas this is essentially a newtype.